<!-- DESCRITPION -->
<launch>
    <node pkg="ira_laser_tools" name="laserscan_multi_merger" type="laserscan_multi_merger" output="screen">
	<param name="destination_frame" value="robot_base_link"/>
	<param name="cloud_destination_topic" value="/merged_cloud"/>
	<param name="scan_destination_topic" value="/scan_multi"/>
	<param name="laserscan_topics" value ="/robot/front_laser/scan /robot/rear_laser/scan" /> 
	<param name="angle_min" value="-3.141592654"/>
	<param name="angle_max" value="3.141592654"/>
	<param name="angle_increment" value="0.008726666681468487"/>
	<param name="scan_time" value="0.0"/>
	<param name="range_min" value="0.10000000149011612"/>
	<param name="range_max" value="30.0"/>
	</node>
	<node name="control" pkg="robotics_scripts" type="robot_control.py" output="screen"></node>
    <node name="combine_lasers_and_odom" pkg="robotics_scripts" type="combiner.py" output="screen"></node>
	<node name="mapping" pkg="robotics_scripts" type="mapping.py" output="screen">
        <param name="sensor_model_p_occ"        value="0.75"/>
        <param name="sensor_model_p_free"       value="0.45"/> 
        <param name="sensor_model_p_prior"      value="0.5"/> 
        <param name="robot_frame"               value="robot_base_link"/> 
        <param name="map_frame"                 value="robot_map"/> 
        <param name="map_center_x"              value="-50.0"/> 
        <param name="map_center_y"              value="-50.0"/> 
        <param name="map_size_x"                value="100.0"/> 
        <param name="map_size_y"                value="100.0"/> 
        <param name="map_resolution"            value="0.1"/> 
        <param name="map_publish_freq"          value="2"/> 
        <param name="update_movement"           value="0.1"/> 
    </node>
    <node name="slam_kf" pkg="robotics_scripts" type="slam_kf.py" output="screen">
        <param name="sensor_model_p_occ"        value="0.75"/>
        <param name="sensor_model_p_free"       value="0.45"/> 
        <param name="sensor_model_p_prior"      value="0.5"/> 
        <param name="robot_frame"               value="robot_base_link"/> 
        <param name="map_frame"                 value="robot_map"/> 
        <param name="map_center_x"              value="-20.0"/> 
        <param name="map_center_y"              value="-20.0"/> 
        <param name="map_size_x"                value="40.0"/> 
        <param name="map_size_y"                value="40.0"/> 
        <param name="map_resolution"            value="0.08"/> 
        <param name="map_publish_freq"          value="1"/> 
        <param name="update_movement"           value="0.1"/> 
    </node>
    <node name="fastSlam" pkg="robotics_scripts" type="fastslam.py" output="screen">
        <param name="sensor_model_p_occ"        value="0.75"/>
        <param name="sensor_model_p_free"       value="0.45"/> 
        <param name="sensor_model_p_prior"      value="0.5"/> 
        <param name="robot_frame"               value="robot_base_link"/> 
        <param name="map_frame"                 value="robot_odom"/> 
        <param name="map_center_x"              value="-20.0"/> 
        <param name="map_center_y"              value="-20.0"/> 
        <param name="map_size_x"                value="40.0"/> 
        <param name="map_size_y"                value="40.0"/> 
        <param name="map_resolution"            value="0.08"/> 
        <param name="map_publish_freq"          value="1"/> 
        <param name="update_movement"           value="0.08"/> 
        <param name="particles"                 value="1"/> 

    </node>
</launch>